packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -37,9 +37,63 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     double maxHeight,
     int imageQuality,
     CameraDevice preferredCameraDevice = CameraDevice.rear,
-  }) {
+  }) async {
     String capture = computeCaptureAttribute(source, preferredCameraDevice);
-    return pickFile(accept: _kAcceptImageMimeType, capture: capture);
+    return pickFile(
+      accept: _kAcceptImageMimeType,
+      capture: await _resizeImage(
+        capture,
+        maxWidth.toInt(),
+        maxHeight.toInt(),
+        imageQuality,
+      ),
+    );
+  }
+
+  static Future<String> _resizeImage(
+    String src,
+    int maxWidth,
+    int maxHeight,
+    int imageQuality,
+  ) {
+    final completer = Completer<String>();
+    final img = html.ImageElement();
```

Comment:
Don't you need to add this `img` to the DOM for the events to fire?

Reply:
I'm haven't had many experience of developing flutter web plugins and I don't know how to proceed here.
Could you elaborate on this @ditman please?

Reply:
Hm, some times for some DOM elements to fire their load/error events they must be injected into the markup of the page (anywhere, but they need to be injected).

I'm not sure this is the case, but I'd test this code across all main browsers (Edge/Safari/Firefox/Chrome) to ensure that the on*** events of the ImageElement are firing as expected.

Same approach for the comment below where we're resetting the image to something blank, and then the desired source again :)

Reply:
@ditman , you are right, events are generally fired if the element is added to DOM tree but while converting my JavaScript code to Dart, I realized that it worked without adding it to DOM for Chrome. I haven't tested for other mentioned browsers.

Reply:
OK I understand what you're saying here, but I don't know how to exactly add this element to the DOM tree :/

I'd need some help here @postacik @ditman .